lib/active_merchant/billing/gateways/garanti.rb

```diff
@@ -251,6 +251,10 @@ def success?(response)
         response[:message] == "Approved"
       end
 
+      def strip_invalid_xml_chars(xml)
+        xml.gsub(/&(?!(?:[a-z]+|#[0-9]+|x[a-zA-Z0-9]+);)/, '&amp;')
```

Comment:
This regex is a bit hard to parse - would the regex [here](http://stackoverflow.com/a/17237801/240418) do the same job but be more clear?